src/libsync/abstractnetworkjob.cpp

```diff
@@ -315,4 +332,48 @@ QString errorMessage(const QString& baseError, const QByteArray& body)
     return msg;
 }
 
+QByteArray requestVerb(const QNetworkReply& reply)
+{
+    switch (reply.operation()) {
+    case QNetworkAccessManager::HeadOperation:
+        return "HEAD";
+    case QNetworkAccessManager::GetOperation:
+        return "GET";
+    case QNetworkAccessManager::PutOperation:
+        return "PUT";
+    case QNetworkAccessManager::PostOperation:
+        return "POST";
+    case QNetworkAccessManager::DeleteOperation:
+        return "DELETE";
+    case QNetworkAccessManager::CustomOperation:
+        return reply.request().attribute(QNetworkRequest::CustomVerbAttribute).toByteArray();
+    case QNetworkAccessManager::UnknownOperation:
+        break;
+    }
+    return QByteArray();
+}
+
+QString networkReplyErrorString(const QNetworkReply& reply)
+{
+    QString base = reply.errorString();
+    int httpStatus = reply.attribute(QNetworkRequest::HttpStatusCodeAttribute).toInt();
+    QString httpReason = reply.attribute(QNetworkRequest::HttpReasonPhraseAttribute).toString();
+
+    // Only adjust HTTP error messages of the expected format.
+    if (httpReason.isEmpty() || httpStatus == 0 || !base.contains(httpReason)) {
+        return base;
+    }
+
+    return AbstractNetworkJob::tr("Server replied \"%1 %2\" to \"%3 %4\"").arg(
+            QString::number(httpStatus),
+            httpReason,
+            requestVerb(reply),
+#if QT_VERSION < QT_VERSION_CHECK(5, 0, 0)
+            reply.request().url().toString()
+#else
+            reply.request().url().toDisplayString()
```

Comment:
@ogoffart For the error list in the ui we don't need the URL, but it might be handy for errors of other network jobs. I had similar doubts and that's why I put it at the end, but would err on the side of including it.